DuplicateAssemblyScanner/DuplicateAssemblyScanner/Settings.cs

```diff
@@ -0,0 +1,124 @@
+namespace DuplicateAssemblyScanner {
+    using ColossalFramework.UI;
+    using DuplicateAssemblyScanner.Structs;
+    using DuplicateAssemblyScanner.Util;
+    using ICities;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Generates the settings screen based on cached results from the assembly scanner.
+    /// </summary>
+    public class Settings {
+        // Markers used in checkbox labels
+        private const string MARKER_BLANK = " ";
+        private const string MARKER_ENABLED = "*";
+
+        // Cache of scan results
+        private static Dictionary<string, int> duplicates_;
+        private static Dictionary<string, List<ModAssembly>> occurrences_;
+        private static bool duplicatesFound_;
+
+        /// <summary>
+        /// Generate the options screen listing all the duplicates (if found).
+        /// </summary>
+        /// 
+        /// <param name="helper">UI helper for building settings screen.</param>
+        public static void CreateUI(UIHelperBase helper) {
+
+            UIHelperBase group;
+
+            if (CacheScan(
+                out Dictionary<string, int> duplicates,
+                out Dictionary<string, List<ModAssembly>> occurrences)) {
+
+                group = helper.AddGroup("Duplicate assemblies were detected.");
+
+                Disable((UICheckBox)group.AddCheckbox("= Assembly is loaded in to app domain", true, NoOp));
+                Disable((UICheckBox)group.AddCheckbox("= Assembly not loaded", false, NoOp));
+
+                foreach (KeyValuePair<string, int> duplicate in duplicates) {
+
+                    if (duplicate.Value > 1) {
+
+                        if (occurrences.TryGetValue(duplicate.Key, out List<ModAssembly> mods)) {
+
+                            RenderDetails(helper, duplicate.Key, duplicate.Value, mods);
+                        }
+                    }
+                }
+            } else {
+                helper.AddGroup("No duplicates.");
+            }
+        }
+
+        /// <summary>
+        /// A dummy click handler for checkboxes. Does nothing.
+        /// </summary>
+        /// 
+        /// <param name="_">Ignored paramter.</param>
+        internal static void NoOp(bool _) {
```

Comment:
The parameter names should not be presented as discards (underscore).
Instead of a dummy method, just provide a lambda that does nothing: dummy empty methods in a class are considered as design flaw.

Reply:
Removed `NoOp` method and replaced event handlers with with `default` which I believe creates empty lambda based on delegate definition.

Reply:
Ok, didn't work in-game, replaced with empty lambda.

Reply:
`default` will always generate a `null` for reference types. Lambdas are reference types.